packages/core/src/components/modal/modal.tsx

```diff
@@ -137,17 +137,18 @@ export class Modal {
     if (event.target !== this.dialog) {
       return;
     }
+    if (!this.isClickInsideDialog && this.closeOnBackdropClick) {
+      this.dismissModal();
+    }
+  }
 
+  private onMouseDown(event: MouseEvent) {
     const rect = this.dialog!.getBoundingClientRect();
-    const isClickOutside =
+    this.isClickInsideDialog =
```

Comment:
Do we even need to check the dimensions, since we konw the event comes from dialog element?